Add 5 laboratory work

Starik1500 opened the pull request:
Add 5 laboratory work

Starik1500 committed:
Add 5 laboratory work

Add 5 laboratory work

Starik1500 committed:
Update and rename README.md to Task for laboratory work

Add a task to the laboratory work and rename README.md to Task for laboratory work

lettaaaaa commented on Laba 5.py:
```diff
@@ -0,0 +1,77 @@
+        self.type = candy_type
+
+    def ate(self):
+        if self.mass * self.amount > 2000:
```
магічне число 2000

Starik1500 replied:
додав коментар

zveres-ss replied:
краще 2000 зробити константою

lettaaaaa commented on Laba 5.py:
```diff
@@ -0,0 +1,77 @@
+    def add_candy(self, candy):
+        self.candies.append(candy)
+
+    def find_the_most_expensive_candies(self, n):
```
незрозуміла назва перемінної 'n'

Starik1500 replied:
перейменував на numbers

lettaaaaa commented on Laba 5.py:
```diff
@@ -0,0 +1,77 @@
+from enum import Enum
+
```
має бути 2 отступи

Starik1500 replied:
> має бути 2 отступи

додав

lettaaaaa commented on Laba 5.py:
```diff
@@ -0,0 +1,77 @@
+    BUTTON = 2
+    POPCORN = 3
+    GUM = 4
+
```
перед класом 2 отступи

Starik1500 replied:
> перед класом 2 отступи

додав

lettaaaaa commented on Laba 5.py:
```diff
@@ -0,0 +1,77 @@
+    GUM = 4
+
+class Candy:
+    def __init__(self, name, mass, amount, price, candy_type):
```
має бути отступ перед петодом

Starik1500 replied:
> має бути отступ перед петодом

додав

lettaaaaa commented on Laba 5.py:
```diff
@@ -0,0 +1,77 @@
+        self.amount = amount
+        self.price = price
+        self.type = candy_type
+
```
перед методом 2 отступи

Starik1500 replied:
додав

lettaaaaa reviewed (changes requested):
Перед усіма класами має бути 2 отступи, та перед методами також має бути 2 оступи

lettaaaaa commented on Laba 5.py:
```diff
@@ -0,0 +1,77 @@
+    GUM = 4
+
+class Candy:
+    def __init__(self, name, mass, amount, price, candy_type):
```
забагато аргументів у класі, має бути не більше 5

Starik1500 replied:
> забагато аргументів у класі, має бути не більше 5

видалив лишній аргумент та пов'язане з ним

Starik1500 committed:
Виправлено код 5 лаби, згідно зауважень

Зробив по 2 відступи, додав коментар до "магічного числа" 2000, та перейменував змінну n на numbers

zveres-ss commented on Laba 5.py:
```diff
@@ -0,0 +1,94 @@
+
+
+def main():
+    Candy1 = Candy("Roshetto", 50, 5, 10)
+    Candy2 = Candy("M&M's", 100, 50, 50)
+    Candy3 = Candy("Orbit", 50, 5, 25)
+    Candy4 = Candy("Eclips", 50, 5, 30)
```
python convention? variable_name_looks_like_this_string
and you have candy name, use this name as variable name

zveres-ss commented on Laba 5.py:
```diff
@@ -0,0 +1,94 @@
+    print(Candy4.get_name())
+
+    
+    Dinner1 = Dinner("Monday", "8:00", [Candy1, Candy2, Candy4])
+    Dinner2 = Dinner("Friday", "15:00", [Candy3])
+    Dinner3 = Dinner("Tuesday", "20:00")
```
monday_dinner
friday_dinner
tuesday_dinner

zveres-ss commented on Laba 5.py:
```diff
@@ -0,0 +1,94 @@
+    Dinner3.add_candy(Candy2)
+
+    
+    expensive_candies = Dinner1.find_the_most_expensive_candies(3)
```
here is a point why convention matters: for any experienced reader `Dinner1.find_the_most_expensive_candies(3)` means a call to static method from class Dinner1
